tutorials/transparent-proxy-and-filtering-on-k8s/index.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Transparent Proxy and Filtering on Kubernetes
+description: Learn how to run a transparent proxy on Kubernetes to filter and intercept traffic out of your deployments.
+author: danisla
+tags: Kubernetes mitmproxy proxy
+date_published: 2017-09-23
+---
+
+Dan Isla | Google Cloud Solution Architect | Google
+
+There are many application environments and scenarios where you may want to filter and intercept all HTTP/S traffic out of a pod. Example use cases include isolating external access to specific HTTP or HTTPS paths, methods and in-flight altering of requests.
+
+An example of HTTPS interception would be filtering access to Google Cloud Storage buckets. Most proxy solutions do not support HTTPS inspection so the filtering that can be done is limited at the IP/DNS level. For the case of a Cloud Storage bucket, the bucket name is embedded under the https://storage.googleapis.com/BUCKET_NAME/object URL so filtering is limited to all buckets under the storage.googleapis.com domain, not any specific bucket.
+
+Another example would be the in-flight modification of requests. If you wanted to transparently add request/response headers to requests between services for tracing purposes, this can be done without the service having to explicitly set them when the request is made.
+
+[Mitmproxy](https://mitmproxy.org/) is an Open Source tool that can be used to intercept and modify HTTP and HTTPS requests transparently using the Python scripting language.
+
+This tutorial uses the [tproxy-sidecar](https://github.com/danisla/kubernetes-tproxy/tree/master/sidecar) init container to create firewall rules in the pod network to block egress traffic out of the pod. The [tproxy-podwatch](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-podwatch) controller watches for pod changes containing the `"initializer.kubernetes.io/tproxy": "true"` annotation and automatically add/removes the local firewall REDIRECT rules to apply the transparent proxy to the pod.
+
+![architecture diagram](./tproxy_diagram.png)
+
+**Figure 1.** transparent proxy architecture diagram
+
+## Objectives
+
+- Create a Kubernetes cluster with Google Container Engine
+- Deploy the tproxy and the tproxy-podwatch pods using Helm
+- Deploy example apps to test external access to a Cloud Storage bucket.
+
+## Before you begin
+
+This tutorial assumes you already have a GCP account and are familiar with the high level concepts of Kubernetes Pods and Deployments.
+
+## Costs
+
+This tutorial uses billable components of GCP, including:
+
+- [Google Container Engine](https://cloud.google.com/container-engine/pricing)
+
+Use the [Pricing Calculator](https://cloud.google.com/products/calculator/#id=f52c2651-4b02-4da3-b8cd-fdbca6ad89a9) to estimate the costs for your environment. The resources used in this tutorial are approximately $2.39 per day.
+
+## Checkout the source repository
+
+1. Open [Cloud Shell](https://console.cloud.google.com/cloudshell)
+
```

Comment:
I suggest opening Cloud Shell in a new tab. For me, it was hard to find my way back to the tutorial. {:target="_blank"}